File: 1776-minimum-operations-to-reduce-x-to-zero/1776-minimum-operations-to-reduce-x-to-zero.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int x) {
        int tar=0, n=nums.size();
        for(const int &e: nums)
            tar+=e;
        tar -= x;
        if(tar==0)
            return n;
        int l=0, curr=0, maxLen=0;
        for(int r=0; r<nums.size(); r++) {
            curr+=nums[r];
            while(l<=r && curr>tar) {
                curr-=nums[l++];
            }
            if(tar==curr)
                maxLen = max(maxLen, r-l+1);
        }
        return maxLen ? n-maxLen : -1;
    }
};

File: 1776-minimum-operations-to-reduce-x-to-zero/1776-minimum-operations-to-reduce-x-to-zero.py
class Solution:
    def minOperations(self, nums: List[int], x: int) -> int:
        tar, n = 0, len(nums)
        for num in nums:
            tar += num
        tar -= x
        if(tar==0):
            return n
        l, maxLen, curr = 0, 0, 0
        for r in range(len(nums)):
            curr += nums[r]
            while(l<=r and curr>tar):
                curr -= nums[l]
                l+=1
            if curr==tar:
                maxLen = max(r-l+1, maxLen)
        return n-maxLen if maxLen!=0 else -1